observable.js

```diff
@@ -444,11 +501,13 @@ const [Observable, Subscriber] = (() => {
           // 6.7. Add the following abort algorithm to subscriber’s subscription controller’s signal:
           subscriber.signal.addEventListener("abort", () => {
             // 6.7.1. Run AsyncIteratorClose(iteratorRecord, NormalCompletion(subscriber’s subscription controller’s abort reason)).
-            if (typeof iteratorRecord.return !== "function") return;
-            const returnResult = iteratorRecord.return(subscriber.signal.reason);
-            if (returnResult === null || typeof returnResult !== "object") {
+            if (typeof iteratorRecord.return !== "function" || done) return;
+            const returnPromise = pTry(() => iteratorRecord.return(subscriber.signal.reason));
+            returnPromise.then((result) => {
+              if (result === null || typeof result !== "object") {
                 throw new TypeError("Iterator .return() must return an Object");
-            }
+              }
+            });
```

Comment:
if this `TypeError` is thrown within the `returnPromise`, how can we pass the test `from(): Sync iterable: error thrown from IteratorRecord#return() can be synchronously caught`

Reply:
isn't that for sync iterables? this part is for async iterables 🤔

Reply:
i'm also not sure how that can be passed - we throw the error from inside the abort event listener, which doesn't seem catchable